fuel-core-interfaces/src/db.rs

```diff
@@ -1,7 +1,91 @@
-use crate::common::fuel_vm::prelude::InterpreterError;
+pub use crate::common::fuel_vm::storage::{
+    ContractsAssets,
+    ContractsInfo,
+    ContractsRawCode,
+    ContractsState,
+};
+use crate::{
+    common::{
+        fuel_storage::Mappable,
+        fuel_tx::{
+            Transaction,
+            UtxoId,
+        },
+        fuel_types::{
+            Address,
+            Bytes32,
+            MessageId,
+        },
+        fuel_vm::prelude::InterpreterError,
+    },
+    model::{
+        Coin,
+        ConsensusId,
+        DaBlockHeight,
+        Message,
+        ValidatorId,
+        ValidatorStake,
+    },
+    relayer::StakingDiff,
+};
 use std::io::ErrorKind;
 use thiserror::Error;
 
+/// The storage table of coins. Each [`Coin`](crate::model::Coin) is represented by unique `UtxoId`.
+pub struct Coins;
+
+impl Mappable for Coins {
+    type Key = UtxoId;
+    type SetValue = Coin;
+    type GetValue = Self::SetValue;
+}
+
+/// The storage table of bridged Ethereum [`Message`](crate::model::Message)s.
+pub struct Messages;
+
+impl Mappable for Messages {
+    type Key = MessageId;
+    type SetValue = Message;
+    type GetValue = Self::SetValue;
+}
+
+/// The storage table of confirmed transactions.
+pub struct Transactions;
+
+impl Mappable for Transactions {
+    type Key = Bytes32;
+    type SetValue = Transaction;
+    type GetValue = Self::SetValue;
+}
+
+/// The storage table of delegate's indexes used by relayer.
+/// Delegate index maps delegate `Address` with list of da block where delegation happened.
+pub struct DelegatesIndexes;
+
+impl Mappable for DelegatesIndexes {
+    type Key = Address;
+    type SetValue = [DaBlockHeight];
+    type GetValue = Vec<DaBlockHeight>;
+}
+
+/// The storage table of relayer validators set.
+pub struct ValidatorsSet;
+
+impl Mappable for ValidatorsSet {
+    type Key = ValidatorId;
+    type SetValue = (ValidatorStake, Option<ConsensusId>);
```

Comment:
Maybe unrelated to this PR, but why is consensus id optional? Every validator must have a consensus id in order to participate in the validation.

Reply:
Maybe when we will have many validators in the future, only part of them will participate in the consensus. The other will idle, waiting for their consensus epoch.

Reply:
I don't think we should make the ConsensusId optional either. Otherwise it becomes very difficult to vote or participate in any sort of validator set activation changes if the consensus key isn't available.

Reply:
nm this was already optional before this PR. We'll revisit all this once we actually start the PoS milestone.